Sofa/framework/Helper/src/sofa/helper/system/PluginManager.h

```diff
@@ -155,27 +157,37 @@ class SOFA_HELPER_API PluginManager
     /// Returns "_d" in debug configuration and an empty string otherwise 
     static std::string getDefaultSuffix();
 
+    enum class PluginLoadStatus : unsigned char
+    {
+        SUCCESS,
+        ALREADY_LOADED,
+        PLUGIN_FILE_NOT_FOUND,
+        INVALID_LOADING,
+        MISSING_SYMBOL,
+        INIT_ERROR
+    };
+
     
     /// Loads a plugin library in process memory. 
     /// @param plugin Can be just the filename of the library to load (without extension) or the full path
     /// @param suffix An optional suffix to apply to the filename. Defaults to "_d" with debug builds and is empty otherwise.
     /// @param ignoreCase Specify if the plugin search should be case insensitive (activated by default). 
     ///                   Not used if the plugin string passed as a parameter is a full path
     /// @param errlog An optional stream for error logging.
-    bool loadPlugin(const std::string& plugin, const std::string& suffix = getDefaultSuffix(), bool ignoreCase = true, bool recursive = true, std::ostream* errlog = nullptr);
-    
+    PluginLoadStatus loadPlugin(const std::string& plugin, const std::string& suffix = getDefaultSuffix(), bool ignoreCase = true, bool recursive = true, std::ostream* errlog = nullptr);
+
     /// Loads a plugin library in process memory. 
     /// @param path The full path of the plugin to load
     /// @param errlog An optional stream for error logging.
-    bool loadPluginByPath(const std::string& path, std::ostream* errlog= nullptr);
+    PluginLoadStatus loadPluginByPath(const std::string& path, std::ostream* errlog= nullptr);
```

Comment:
I noticed that the PluginManager.h::loadPluginByPath is using different c++ syntax for the same method signature and wonder if this is something intended ?  
```cpp
// in PluginManager.h it is: 
PluginLoadStatus PluginManager::loadPluginByPath(...);

// in PluginManager.cpp it is: 
auto PluginManager::loadPluginByPath(...) -> PluginLoadStatus
```

Reply:
I find a bit confusing using the two signature instead of uing auto XXX -> PluginLoadStatus everywhere. But as long as IDEs can track this is the same i will get used to that if needed :)

Reply:
Alright, I can understand. I made the change to have a unique signature syntax.